src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10055,6 +10062,29 @@ private bool ScanParenthesizedImplicitlyTypedLambda(Precedence precedence)
             return false;
         }
 
+        private bool IsParenVarCommaSyntax()
+        {
+            var next = this.PeekToken(1);
+            var afterKind = this.PeekToken(2).Kind;
+
+            // Ensure next token is a variable
+            if (next.Kind == SyntaxKind.IdentifierToken)
+            {
+                if ((!this.IsInQuery || !IsTokenQueryContextualKeyword(next))) {
+                    // Variable must be directly followed by a comma if not followed by postfix
+                    if (afterKind == SyntaxKind.CommaToken)
+                    {
+                        return true;
+                    }
+                    else if ((afterKind == SyntaxKind.ExclamationToken && this.PeekToken(3).Kind == SyntaxKind.CommaToken))
```

Comment:
if you just make this a normal 'if' you can easily add a comment statin the case this is checking for.  (Same with the case above this as well).

Reply:
also, extra parens.  An `if` statemnet already has the surrounding parens for the conditional.  We don't need an immediately contained parenthesized exprssion *within* that.